tests/test_cli_compile.py

```diff
@@ -2457,6 +2447,38 @@ def test_triple_equal_pinned_dependency_is_used(
 )
 
 
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_not_specified_input_file(
+    runner, make_module, fname, content, tmpdir, monkeypatch
+):
+    """
+    It should raise an error if there are no input files or default input files
+    such as "setup.py", "requirements.in" or "pyproject.toml".
+    """
+    make_module(fname=fname, content=content)
+
+    monkeypatch.chdir(tmpdir)
+    out = runner.invoke(cli)
+    monkeypatch.undo()
+
+    assert "small-fake-a==0.1" in out.stderr
```

Comment:
```
(Pdb++) print(out.stderr)`
  ERROR: Could not find a version that satisfies the requirement small-fake-a==0.1 (from sample-lib (pyproject.toml)) (from versions: none)
```

>     It should raise an error if there are no input files or default input files
>     such as "setup.py", "requirements.in" or "pyproject.toml".

Could you elaborate on which exact error should be raised?

Reply:
> Could you elaborate on which exact error should be raised?

Not really -- this docstring was present in the test before my changes, I just expanded the list of expected files. Perhaps it can be rewritten with more details.

Reply:
This test does not really differ from the other test. Maybe drop this test completely?

Reply:
Ah I see you are talking about the test when the files are present; yeah, that docstring should not be there at all, it was duplicated my mistake.

Reply:
We need to test two separate functionalities when calling `pip-compile` without arguments: one that tests that a file is parsed if present, and the other one that that tests that an error is raised if none of the default files are present.

Reply:
> Test that a default-named file is parsed if present.

I don't understand why there is an error checking in the test given the description above.

Reply:
You're right, it shouldn't; but for some reason it's returning status code 1. 

This test, `test_not_specified_input_file`, is supposed to work exactly the same as `test_input_formats`, with the only difference being that in this one we don't pass the file path; I have literally copied it and changed the CLI invocation not to pass the filename (apparently that use case was not even tested before).

I'm trying to figure out why is it returning 1 now. 🤔

Reply:
I see. That's essentially `test_input_formats` except `monkeypatch.chdir(tmpdir)`. Why do we change the current directory by the way?

Reply:
Because the `make_module` fixture creates the files in the `tmpdir`, and we are testing the functionality which looks for the files in the CWD (see earlier discussion). So our options are:

1. Patch `make_module` to be able to create files in CWD; I did that before but reverted as recommended.
2. Create the file in `tmpdir` and change the CWD to `tmpdir` (current approach).
3. Create the file in `tmpdir`, copy it to CWD and delete after the test; I might try that next. 🤷‍♂️

Reply:
> Because the make_module fixture creates the files in the tmpdir

I see. Let's make it explicit then by:

```python
module_path = make_module(fname=fname, content=content)
monkeypatch.chdir(module_path / '..')
```

Reply:
OK, I think I figured it out; I was a bit too optimistic with removing all the invocation params, rather than just the filename. I also changed the test to use `chdir` to the file's actual `dirname` instead of using `tmpdir`.

Reply:
Thanks! That looks nice now!

Reply:
> Let's make it explicit then

Yeah - I used `os.path` rather than `pathlib.Path` as it has been used in other tests, for consistency; I personally prefer the latter.

Reply:
Me too. We can refactor this in a following up PR.